Google.Api.Gax.Grpc/Rest/RestChannel.cs

```diff
@@ -188,14 +188,60 @@ internal Status GetStatus()
             {
                 var grpcStatus = OriginalResponseMessage.StatusCode switch
                 {
-                    HttpStatusCode.OK => StatusCode.OK,
-                    HttpStatusCode.Unauthorized => StatusCode.Unauthenticated,
-                    HttpStatusCode.Forbidden => StatusCode.PermissionDenied,
+                    // Note: this switch statement contains all the unique named values within HttpStatusCode (as of 2021-08-09).
+
+                    HttpStatusCode.Accepted => StatusCode.OK,
+                    HttpStatusCode.Ambiguous => StatusCode.Unknown,
+                    HttpStatusCode.BadGateway => StatusCode.InvalidArgument,
                     HttpStatusCode.BadRequest => StatusCode.InvalidArgument,
+                    // Note: this is ambiguous between FailedPrecondition, Aborted or Unavailable.
```

Comment:
Should we go with Aborted so that clients maybe retry (Aborted is the one that's usually OK to retry).
On the other hand if it's really a FailedPrecondition (as in it's an erroneous argument) retrying won't solve it. But still client code should retry only a few times, etc.

So, I would favour giving client code the opportunity to recover by retrying.

Reply:
Java uses different error codes based on the message. (Not entirely happy with that, really.) We'll see what consensus we get. I've made a note in the code.